plotly/tools.py

```diff
@@ -398,60 +400,464 @@ def get_subplots(rows=1, columns=1, horizontal_spacing=0.1,
         fig['data'] += [Scatter(x=[1,2,3], y=[2,1,2], xaxis='x2', yaxis='y2')]
 
     Example 2:
+        # subplots with shared x axes
+        fig = tools.get_subplots(rows=2, shared_xaxes=True)
+        fig['data'] += [Scatter(x=[1,2,3], y=[2,1,2], yaxis='y1')]
+        fig['data'] += [Scatter(x=[1,2,3], y=[2,1,2], yaxis='y2')]
+
+    Example 3:
+        # irregular subplot layout
+        fig = tools.get_subplots(rows=2, columns=2,
+                                 specs=[[{}, {}], [{'colspan': 2}]])
+        fig['data'] += [Scatter(x=[1,2,3], y=[2,1,2], xaxis='x1', yaxis='y1')]
+        fig['data'] += [Scatter(x=[1,2,3], y=[2,1,2], xaxis='x2', yaxis='y2')]
+        fig['data'] += [Scatter(x=[1,2,3], y=[2,1,2], xaxis='x3', yaxis='y3')]
+
+    Example 4:
+        # insets
+        fig = tools.get_subplots(insets=[{'cell':(0,0), 'l': 0.7, 'b': 0.3}])
+        fig['data'] += [Scatter(x=[1,2,3], y=[2,1,2])]
+        fig['data'] += [Scatter(x=[1,2,3], y=[2,1,2], xaxis='x2', yaxis='y2')]
+
+    Example 5:
         # print out string showing the subplot grid you've put in the layout
         fig = tools.get_subplots(rows=3, columns=2, print_grid=True)
 
-    key (types, default=default):
-        description.
+    Keywords arguments with constant defaults:
+
+    rows (kwarg, int, default=1):
+        Number of rows on the figure.
+
+    columns (kwarg, int, default=1):
+        Number of columns on the figure.
+
+    shared_xaxes (kwarg, boolean or list, default=False)
+        Assign shared x axes.
+        If True, all x axes are shared.
+        To assign shared x axes per subplot grid cell (see 'specs'),
+        send list (or list of lists, one list per shared axis)
+        of cell index tuples.
+
+    shared_yaxes (kwarg, boolean or list, default=False)
+        Assign shared y axes.
+        If True, all y axes are shared.
+        To assign shared y axes per subplot grid cell (see 'specs'),
+        send list (or list of lists, one list per shared axis)
+        of cell index tuples.
+
+    start_cell (kwarg, 'bottom-left' or 'top-left', default='bottom-left')
+        Choose the starting cell in the subplot grid used to set the
+        domains of the subplots.
+
+    print_grid (kwarg, boolean, default=False):
+        If True, prints a tab-delimited string representation of
+        your plot grid.
+
+    Keyword arguments with variable defaults:
+
+    horizontal_spacing (kwarg, float in [0,1], default=0.2 / columns):
+        Space between subplot columns.
 
-    rows (int, default=1):
-        Number of rows, evenly spaced vertically on the figure.
+        Applies to all columns (use 'specs' subplot-dependents spacing)
 
-    columns (int, default=1):
-        Number of columns, evenly spaced horizontally on the figure.
+    vertical_spacing (kwarg, float in [0,1], default=0.3 / rows):
+        Space between subplot rows.
+        Applies to all rows (use 'specs' subplot-dependents spacing)
```

Comment:
:+1:  We should mention `vertical_spacing` and `horizontal_spacing` in the `specs` doc string.